src/edge/recombine.py

```diff
@@ -538,7 +538,13 @@ def recombine_region(genome, region, min_homology_arm_length, op, new_fragment_d
         new_fragment_dict[f.id] = 1
 
         replaced = 0
-        if region_start <= region_end:
+
+        # added "+1 here": region_start and region_end are interpreted as
+        # replacing the bp starting at region_start and ending at region_end.
+        # If region_end is region_start-1, that means it's just an insertion
+        # and we should allow that.
+
+        if region_start <= region_end + 1:
```

Comment:
Admittedly I'm a complete novice to this codebase, so take this with a grain of salt. But I got hung up on why/how we'd ever have to increment `region_end` by 1 for it to equal `region_start`. It just feels like if anything, we might want `region_start + 1 <= region_end`.

I suppose your comment explains the special case for insertion, that -- to use concrete numbers -- `region_end` could be bp 8 and `region_start` would be bp 9, and that's then an insertion. Still feels weird; I'd expect 9 and 9.

Anyway. My instinct to address this, and hopefully it doesn't just obfuscate the concern, is to introduce another variable:
```
length_to_recombine = region_end - region_start + 1
if length_to_recombine >= 0:
    f.replace_bases(region_start, length_to_recombine, cassette)
    replaced = length_to_recombine
```

Reply:
The region is the segment of bps to be removed. The first bp of the region, along with the last bp, should be removed. Hence, region_start is the first bp to be removed, and region_end is the last bp to be removed. When region_start == region_end, we are removing exactly 1 bp. When region_start = region_end+1, it's when we are not removing anything.

I like your length_to_recombine variable idea, but I will call it number_of_bps_to_remove!

Reply:
Thanks for humoring me! Your explanation makes sense, too.